src/server/req_holdjob.c

```diff
@@ -210,6 +210,7 @@ req_holdjob(struct batch_request *preq)
 		} else {
 			pjob->ji_qs.ji_svrflags |=
 				JOB_SVFLG_HASRUN | JOB_SVFLG_CHKPT;
+			pjob->ji_qs.ji_svrflags |= JOB_SVFLG_HASHOLD;
```

Comment:
I think this flag can move inside the previous statement itself.

I do have a question, You are adding this flag because job_obit() is unable to identify whether the job it is dealing with was checkpointed or has actually ended. But, my point of confusion is that why does the job holds on to JOB_SVFLG_CHKPT? When it is requeued the first time after checkpointing shouldn't we just remove CHKPT flag from the job.
I'm probably missing something here.